packages/terra-site/tests/jest/theming-plugin-spec.test.jsx

```diff
@@ -0,0 +1,80 @@
+/* globals spyOn */
+/* eslint-disable global-require */
+
+import postcss from 'postcss';
+import fs from 'fs';
+
+describe('theming-plugin', () => {
+  let mockStyles;
+  let pluginKey;
+  let outputtedFileName;
+  let outputtedEncoding;
+  let testingFunction;
+  let ThemingPlugin;
+
+  beforeEach(() => {
+    spyOn(postcss, 'plugin').and.callFake((key, processingFunction) => {
+      pluginKey = key;
+      return processingFunction;
+    });
+
+    spyOn(fs, 'writeFileSync').and.callFake((fileName, hash, encoding) => {
+      outputtedFileName = fileName;
+      outputtedEncoding = encoding;
+      return hash;
+    });
+
+    // We need to require this here as the theming-plugin relies on a function that needs to be stubbed.
+    ThemingPlugin = require('../../theming-plugin');
+    testingFunction = ThemingPlugin();
+    mockStyles = `
+        :local {
+          .testing-component {
+            font-size: var(--terra-font-size, 1.5em);
+            color: var(--terra-color, #000);
+            background-color: var(--terra-background-color, #000);
+            margin: 0.357em;
+          }
+        }
+    `;
+  });
+
+  it('should have the correct key for the plugin', () => {
+    expect(pluginKey).toEqual('theming-plugin');
+  });
+
+  describe('when the SCSS comes from a file', () => {
+    it('should find all of the appropriate css', () => {
+      const rootNode = postcss.parse(mockStyles);
+      const expectedHash = {
+        TestingComponent: {
+          '--terra-font-size': '1.5em',
+          '--terra-color': '#000',
+          '--terra-background-color': '#000',
+        },
+      };
+      rootNode.source = { input: { file: 'TestingComponent.scss' } };
+
+      expect(testingFunction(rootNode)).toEqual(JSON.stringify(expectedHash));
+      expect(outputtedFileName).toEqual('themeable-variables.json');
+      expect(outputtedEncoding).toEqual('utf8');
+    });
+  });
+
+  describe('when the SCSS comes from a file', () => {
```

Comment:
nit: maybe rename to different describes statement than line 46.

Reply:
Done. This is for when the style does not have a file attached to it's source. https://github.com/cerner/terra-core/pull/758/commits/911f4139c72b2ddfaa7c18ff5d472678f05d91ce